packages/@glimmer/reference/test/references-test.ts

```diff
@@ -37,111 +34,284 @@ class TrackedDict<T> {
+      assert.equal(valueForRef(outerRef), 'BAR');
 
-  second.set('foo', 'BAR');
+      assert.strictEqual(count, 2, 'computed');
+    });
 
-  assert.equal(valueReference.value(), 'bar');
-  assert.equal(valueReference.value(), 'bar');
-  assert.equal(valueReference.value(), 'bar');
+    test('can create children of compute refs', assert => {
+      class Child {
+        @tracked value = 123;
+      }
 
-  assert.strictEqual(computed, 1, 'computed');
+      class Parent {
+        @tracked child = new Child();
+      }
 
-  dictReference.update(second);
+      let parent = new Parent();
 
-  assert.equal(valueReference.value(), 'BAR');
-  assert.equal(valueReference.value(), 'BAR');
-  assert.equal(valueReference.value(), 'BAR');
+      let computeRef = createComputeRef(() => parent.child);
+      let valueRef = childRefFor(computeRef, 'value');
 
-  assert.strictEqual(computed, 2, 'computed');
+      assert.equal(valueForRef(valueRef), 123, 'value is correct');
+      assert.equal(valueForRef(valueRef), 123, 'value is correct');
+      assert.equal(getCount, 1, 'get called correct number of times');
 
-  second.set('foo', 'foo-bar');
+      parent.child.value = 456;
 
-  assert.equal(valueReference.value(), 'foo-bar');
-  assert.equal(valueReference.value(), 'foo-bar');
-  assert.equal(valueReference.value(), 'foo-bar');
+      assert.equal(valueForRef(valueRef), 456, 'value updated correctly');
+      assert.equal(valueForRef(valueRef), 456, 'value is correct');
+      assert.equal(getCount, 2, 'get called correct number of times');
 
-  assert.strictEqual(computed, 3, 'computed');
+      assert.equal(isUpdatableRef(valueRef), true, 'childRef is updatable');
+
+      updateRef(valueRef, 789);
+
+      assert.equal(valueForRef(valueRef), 789, 'value updated correctly');
+      assert.equal(getCount, 3, 'get called correct number of times');
+      assert.equal(setCount, 1, 'set called correct number of times');
+
+      parent.child = new Child();
+
+      assert.equal(valueForRef(valueRef), 123, 'value updated correctly when parent changes');
+      assert.equal(getCount, 4, 'get called correct number of times');
+    });
+  });
+
+  module('unbound ref', () => {
+    test('children of unbound refs are not reactive', assert => {
+      class Parent {
+        @tracked child = 123;
+      }
+
+      let parent = new Parent();
+
+      let constRef = createUnboundRef(parent);
+      let childRef = childRefFor(constRef, 'child');
+
+      assert.equal(valueForRef(childRef), 123, 'value is correct');
+
+      parent.child = 456;
+
+      assert.equal(valueForRef(childRef), 123, 'value updated correctly');
+    });
+  });
+
+  module('invokable ref', () => {
+    test('can create invokable refs', assert => {
+      let ref = createComputeRef(
+        () => {},
+        () => {}
+      );
+
+      let invokableRef = createInvokableRef(ref);
+
+      assert.ok(isInvokableRef(invokableRef));
+    });
+  });
+
+  module('read only ref', () => {
```

Comment:
Should test the "normal battery of things", this test assumes details about implementation and it shouldn't.